uplink/piecestore/upload.go

```diff
@@ -181,8 +181,10 @@ func (client *Upload) Commit(ctx context.Context) (_ *pb.PieceHash, err error) {
 
 	// sign the hash for storage node
 	uplinkHash, err := signing.SignUplinkPieceHash(ctx, client.privateKey, &pb.PieceHash{
-		PieceId: client.limit.PieceId,
-		Hash:    client.hash.Sum(nil),
+		PieceId:   client.limit.PieceId,
+		PieceSize: client.offset,
+		Hash:      client.hash.Sum(nil),
+		Timestamp: client.limit.OrderCreation,
```

Comment:
I think this should be just `time.Now()`. Its just timestamp when PieceHash was created.

Reply:
I really tend to dislike adding `time.Now()` deep in code like this, because it makes things nondeterministic and hard to test. This can change in the future without breaking backwards compat (because the storagenodes have to keep track of the exact timestamp sent whatever it is), so I would like to put off that decision if possible, and get the size change in before the wipe.